grabbit/src/main/groovy/com/twcable/grabbit/server/batch/steps/jcrnodes/JcrNodesProcessor.groovy

```diff
@@ -88,10 +90,51 @@ class JcrNodesProcessor implements ItemProcessor<JcrNode, Node> {
             }
         }
 
-        final List<JcrProperty> properties = jcrNode.properties.toList()
-        Node.Builder nodeBuilder = Node.newBuilder()
+        // It checks if the node parent has required primary type as "nt:hierarchyNode"
+        // and then skip it as the node has already been processed by parent node
+        if(isRequiredHierarchyChildNode(jcrNode)) {
```

Comment:
Indeed, but you said it yourself.  Those patterns are creational patterns - JcrNodeFactory wouldn't accurately describe what it *is*, or *does*.  If all the class did was create JCR nodes, then absolutely; but what it *is*, and *does*; is it wraps (or decorates) a JCR node, and gives JCR nodes extra behavior.  The argument I don't think can be made for the command pattern yet, but maybe it will evolve to that